### ISSUE TITLE
[GAS-070] Error _onFormSubmit when have the new request 

### ISSUE THREAD

**Issue by tuanngocptn** [ashtonfei/google-apps-script-projects#2]
Hello I have some problem when running on GAS-070 (https://github.com/ashtonfei/google-apps-script-projects/tree/GAS-070).
.
Some user missing additional value like _uid, _status, _response_id, _approver_1, _approver_2.

So I found the reason. Sometimes, the additional value inserted wrong area. That inserted in the before line.

<img width="525" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/65dc8aa6-2ec9-41a2-97f8-c563b6c8e9f7">

Like the image above UID-00059 must fill in the next line. But it replace the UID-00058 area.

That will make some error of system when we run the application in flow.

Thanks for read. And hope that will fix soon.

**Comment by tuanngocptn** [comment]
Hello @ashtonfei. 
Did you read this issue.
Thanks

**Comment by ashtonfei** [comment]
@tuanngocptn Is it always replace the line above the actual response line? If it only happens to some responses, we'll need more investigation.

**Comment by tuanngocptn** [comment]
Hello Ashtonfei
I'm apply this form for my team. And hope that will fix soon.
Thanks @ashtonfei for response.

**Comment by ashtonfei** [comment]
@tuanngocptn Were the responses 58 and 59 submitted very closely? Like almost the same time?

**Comment by tuanngocptn** [comment]
This problem happens quite often.

<img width="604" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/a4d73162-6e38-4529-9524-ac376f3ccc73">

.

.

And this is error is showing on the Executions report.

<img width="957" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/3e3902cd-09bb-4e66-8bd3-d58048faa00f">

Thanks @ashtonfei

**Comment by tuanngocptn** [comment]
> 

Nope. I don't think so, Cus I checked in the timestamp column. Most of them are not in the same time.

<img width="190" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/1418ad68-7aca-4733-aa6b-1c9496450858">

**Comment by ashtonfei** [comment]
I updated the onOpen function to fix the permission issue on creating triggers. You will need to install the trigger manually.
``` javascript
function onOpen() {
  // const trigger = createTrigger(); // removed
  const ui = FormApp.getUi();
  const menu = ui.createMenu("Approval");
  menu.addItem("Reset UID", "resetUid");
  menu.addItem("Install trigger", "createTrigger"); // added
  menu.addToUi();
}
```

**Comment by tuanngocptn** [comment]
> I updated the onOpen function to fix the permission issue on creating triggers. You will need to install the trigger manually.
> 
> ```js
> function onOpen() {
>   // const trigger = createTrigger(); // removed
>   const ui = FormApp.getUi();
>   const menu = ui.createMenu("Approval");
>   menu.addItem("Reset UID", "resetUid");
>   menu.addItem("Install trigger", "createTrigger"); // added
>   menu.addToUi();
> }
> ```

Hello.

I just added this trigger after changed the code.

<img width="756" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/82c0f348-d761-42d9-9189-bb78eaa845d8">

I will send you the result after several days.

Really thanks @ashtonfei

**Comment by tuanngocptn** [comment]
Hi @ashtonfei 

after several day running.

Seem like problem still there 

Do you have any idea for fix that.

<img width="365" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/22292704/087d6b28-9f81-4630-9d29-7cac02fc1998">

Thanks

**Comment by tuanngocptn** [comment]
Hello @ashtonfei. Are you still following this issue. Thanks

**Comment by ashtonfei** [comment]
@tuanngocptn I checked the code, I think there is a defect with the logic in my original design that the script is always checking the last row when there is a submission. Is response edit allowed in your form? In current logic, the response edit could trigger the script to overwrite the last row in the sheet.

**Comment by tuanngocptn** [comment]
> @tuanngocptn I checked the code, I think there is a defect with the logic in my original design that the script is always checking the last row when there is a submission. Is response edit allowed in your form? In current logic, the response edit could trigger the script to overwrite the last row in the sheet.

Hello @ashtonfei 

```Is response edit allowed in your form?``` => sorry I don't get the point of this question. I just configure like your video in youtube.

```I think there is a defect with the logic in my original design that the script is always checking the last row when there is a submission``` => Can we change to update by timestamp column instead of "last row". I think timestamp column is unique column now.

Thanks

**Comment by ashtonfei** [comment]
If form response edit is enabled, we have will have the issue on the last row when an edit submission happens. We'll need to change the trigger to Spreadsheet On Form Submit, in this type of trigger, we can access the row number which the response is saved to.
<img width="859" alt="image" src="https://github.com/ashtonfei/google-apps-script-projects/assets/16481229/12409fd0-1934-422d-bcbd-d65ae7b29268">

**Comment by tuanngocptn** [comment]
I just checked the Responses setting and It all have off already before. Seem like that is not the core problem
As you mention we need to change the trigger. For updating exactly the row.
So, Can I help you for do something? For dealing with this problem faster.
Thanks @ashtonfei

**Comment by prateek21865** [comment]
Hey @ashtonfei I want to import some other data from a sheet1 which is in the same spreadsheet having form responses 1. I want add 2 -3 data values in the task object. What I have to do.. please suggest me..and after approving the email the data gets reflected to that sheet only where the data is been Imported...
